test/Microsoft.AspNet.Mvc.TagHelpers.Test/ImageTagHelperTest.cs

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNet.FileProviders;
+using Microsoft.AspNet.Hosting;
+using Microsoft.AspNet.Http.Internal;
+using Microsoft.AspNet.Mvc.ModelBinding;
+using Microsoft.AspNet.Mvc.Rendering;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using Microsoft.AspNet.Routing;
+using Microsoft.Framework.Caching;
+using Microsoft.Framework.Caching.Memory;
+using Microsoft.Framework.Runtime;
+
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers
+{
+    public class ImageTagHelperTest
+    {
+
+        [Fact]
+        public void PreservesOrderOfSourceAttributesWhenRun()
+        {
+            // Arrange
+            var context = MakeTagHelperContext(
+                attributes: new TagHelperAttributeList
+                {
+                    { "alt", new HtmlString("alt text") },
+                    { "data-extra", new HtmlString("something") },                    
+                    { "title", new HtmlString("Image title") },
+                    { "src", "testimage.png" },
+                    { "asp-file-version", "true" }
+                });
+            var output = MakeImageTagHelperOutput(
+                attributes: new TagHelperAttributeList
+                {
+                    { "alt", new HtmlString("alt text") },
+                    { "data-extra", new HtmlString("something") },                    
+                    { "title", new HtmlString("Image title") },
+                    { "src", "testimage.png" },
+                    { "asp-file-version", "true" }
```

Comment:
Bit unrealistic.  `src` and `asp-file-version` are _not_ in the `TagHelperOutput.Attributes` collection by default.  Bound attributes appear only in the `TagHelperExecutionContext.AllAttributes` collection when tag helper execution begins.  That's why the `CopyHtmlAttribute()` call is necessary in the `<img>` tag helper.